docusaurus/docs/testing/extensions-test.md

```diff
@@ -0,0 +1,45 @@
+# Extensions Tests
+
+The Extensions framework is tested by the GitHub action `check-plugins.yaml`. This runs the script `./shell/scripts/test-plugins-build.sh`
+which runs the actual tests.
+
+The tests check the following:
+
+- Ensures that Verdaccio is installed and running locally (Verdaccio provides a local NPM registry that the tests can use to publish packages without having to publish to npm)
+- Verifies the build of the following, by building and then publishing them to the local registry
+  - Shell
+  - Yarn Creator packages
+  - Rancher components package
+- Verifies that a new app can be created with the `@rancher/app` creator and built (test case of a new extension in a new repository)
+- Verifies that an extension can be added with the `@rancher/pkg` creator to the new app and that the dashboard build passes with the new extension present
+- Verifies that the extension can be build standalone
+- Verifies that an extension can be added to the existing dashboard and that the dashboard build passes with the new extension present
+
+## Running Tests locally
+
+You can run the script `./shell/scripts/test-plugins-build.sh` locally in a terminal to run the test suite on a developer machine.
+
+Run the script with `-h` to see the list of options available.
+
+## Possible causes of test failures
+
+### Dependencies
+
+For the standalone case of a new app using the `@rancher/shell` package, the dependency versions can end up being different to that of the dashboard codebase. This is typically due to dependencies being top-level dependencies in the case of the dashboard codebase, but child dependencies in the case of the new app. In the later case, yarn can hoist dependency versions differently.
+
+To debug this situation, run the test script with `-i -k` - this will only run the standalone tests (`-i` ignores the in-tree tests) and will keep the generate application after the test run (`-k`).
```

Comment:
I get this when I try to run the command `./shell/scripts/test-plugins-build.sh -i -k`

```
Extensions framework build tests
In-tree tests will not be run
Folder for test appliaction will not be deleted after test run
Verdaccio is already running
You should ensure a user is set up and logged in for publishing
npm ERR! code ENEEDAUTH
npm ERR! need auth This command requires you to be logged in.
npm ERR! need auth You need to authorize this machine using `npm adduser`

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gleo/.npm/_logs/2023-04-25T07_34_36_251Z-debug.log
```

As we use a private registry, this error should not happen and the login should be done by the test.

Using `npm login --registry=http://localhost:4873/` solves this error report.